fixes CPU usage going to 100% whenever PC is resumed from standby on certain
sites like github.com.
the underlying cause is that some (probably XHR) connection is interrupted
by the wake-up but not registered as such in the browser code, and
HasPendingEvent always returns true for that specific fd (which is a pipe).

--- pale-moon.org/platform/netwerk/base/nsSocketTransportService2.cpp
+++ pale-moon/platform/netwerk/base/nsSocketTransportService2.cpp
@@ -486,7 +486,7 @@
         mPollList[0].out_flags = 0;
         pollList = mPollList;
         pollCount = mActiveCount + 1;
-        pollTimeout = pendingEvents ? PR_INTERVAL_NO_WAIT : PollTimeout(ts);
+        pollTimeout = PR_MillisecondsToInterval(100);
     }
     else {
         // no pollable event, so busy wait...
